### PR TITLE
ISSUE #328

### PR DESCRIPTION
**Pull request by jancris100**
✅ Closes https://github.com/undefinedorgcr/gostarkme-backend/issues/328
✅ Added unit tests for legacy map
✅ Update Test Cases for update_receive_donation Method
✅ Added boundary and Edge Cases:

Implemented Test Cases:

✅ Legacy Map Tests:
Verify that a new donator is correctly registered with their initial donation amount.
Verify that an existing donator's total donation is updated upon subsequent donations.

✅ update_receive_donation Method:
Ensure the method updates the legacy map before emitting the donation event.
Validate that the donation event is emitted with correct parameters.

✅ Boundary & Edge Cases:
Tested a new donator contributing a small amount.
Tested an existing donator contributing a large amount.
Verified behavior when multiple donations are made in rapid succession.

Comments
Added a new function on fund.cairo, get_donation_amount
This function retrieves and returns the total amount donated by a specific donor from the legacy map (self.donators).

Current output
image

### CODE REVIEW

**Comment by jancris100** [comment]
@EmmanuelAR I think is ready let me know if you need any request

File: command_output.log [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (EmmanuelAR) [inline command_output.log:1]: Please remove this file Sr

File: tests/test_fund.cairo [hunk 2]
```diff
@@ -75,7 +78,6 @@ fn _setup_() -> ContractAddress {
     calldata.append_serde(CONTACT_HANDLE_1());
     calldata.append_serde(REASON_1());
     calldata.append_serde(FundTypeConstants::PROJECT);
-
```
Comment (EmmanuelAR) [inline tests/test_fund.cairo:78]: Please leave this white lines

File: tests/test_fund.cairo [hunk 3]
```diff
@@ -444,42 +446,34 @@ fn test_set_contact_handle_wrong_owner_or_admin() {
 #[fork("Mainnet")]
 fn test_update_received_donation() {
     let contract_address = _setup_();
-
```
Comment (EmmanuelAR) [inline tests/test_fund.cairo:447]: Please leave this white lines

File: tests/test_fund.cairo [hunk 4]
```diff
@@ -444,42 +446,34 @@ fn test_set_contact_handle_wrong_owner_or_admin() {
...
     let mut spy = spy_events();
-
```
Comment (EmmanuelAR) [inline tests/test_fund.cairo:449]: Please leave this white lines

File: tests/test_fund.cairo [hunk 5]
```diff
@@ -444,42 +446,34 @@ fn test_set_contact_handle_wrong_owner_or_admin() {
...
     let strks: u256 = 500 * ONE_E18;
-
```
Comment (EmmanuelAR) [inline tests/test_fund.cairo:451]: Please leave this white lines

File: tests/test_fund.cairo [hunk 6]
```diff
@@ -444,42 +446,34 @@ fn test_set_contact_handle_wrong_owner_or_admin() {
...
     let dispatcher = IFundDispatcher { contract_address };
     let minter_address = contract_address_const::<StarknetConstants::STRK_TOKEN_MINTER_ADDRESS>();
     let token_address = contract_address_const::<StarknetConstants::STRK_TOKEN_ADDRESS>();
     let token_dispatcher = IERC20Dispatcher { contract_address: token_address };
-
```
Comment (EmmanuelAR) [inline tests/test_fund.cairo:456]: Please leave this all the old white lines

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]
Excellent work Sr, could you please run scarb fmt to format

**Comment by EmmanuelAR** [comment]
@jancris100 I see some conflicts on some files please pull the latest changes on the repo and solve it please Sr

**Comment by jancris100** [comment]
image

**Comment by jancris100** [comment]
Hi i will see the PR tomorrow because need some changes, in the new code change it         donators: LegacyMap::<ContractAddress, u256>,
 to         donators: LegacyMap::<ContractAddress, DonatorInfo>,

**Comment by EmmanuelAR** [comment]
Take your time Sr!

**Comment by EmmanuelAR** [comment]
This Pr will be close because @jancris100 create a new one
